doc/networking.md

```diff
@@ -0,0 +1,132 @@
+# Livepeer Networking Protocol v2 Spec
+
+## High level flow
+
+![broadcaster-transcoder network v2 1](https://user-images.githubusercontent.com/292510/41455677-c8437268-7032-11e8-9ce8-bfdd9b6e3fc0.png)
+[Sequence diagram source](https://sequencediagram.org/index.html#initialData=C4S2BsFMAIBkQG6QA6UgJ2gOUsA7gPboDWIAdgObQIBMAUHcgIbqgDGIzZw0ASpBRABnYOgCedAELoCTACZsmIjAFoAfP0EjxALmgB5dGwAWkbU2BE+kAI51Nw0WJXrpshUuAY9hk2dEWVvxCyAxu8orK6Oq+puaW6DoAOmQAKuhMZEJsBHIY1nbhHlEAPC6x-hkJOumZ2bn5AJJkAGYEDAzgBAShRZFe0Wq1WTl5idAAygIAtpDcBVIyEZ4YZSrD9WN6UxSz88Ghc3J0QA)
+
+## Broadcaster to Registry
+
+The broadcaster does an on-chain lookup to retrieve the orchestrator's ServiceURI in order to initiate an transcoding job.
+
+## Broadcaster to Orchestrator:
+
+### gRPC `GetTranscoder`
+
+This is called by the broadcaster to request information on a transcoder from an orchestrator. The request contains the following data:
+
+```protobuf
+// This request is sent by the broadcaster in `GetTranscoder` to request
+// information on which transcoder to use.
+message TranscoderRequest {
+
+  // ID of the job that the broadcaster needs a transcoder for
+  int64 jobId = 1;
+
+  // Broadcaster's signature over the jobId
+  bytes sig = 2;
+}
+```
+
+The `sig` field is the broadcaster's signature using its Ethereum key, over the jobId expressed as a string. The signature consists of the resulting bytes without any further encoding.
+`sig = broadcaster.Sign(jobId.String())`
+
+Verification of `TranscoderRequest` consists of the following steps:
+1. Fetch the job based on JobId.
+    * TODO distinguish if an orchestrator is lagging behind and hasn't yet received the block with the job. Might require incorporating the node's **current** block number into the request. `CreationBlock`  as stored in the job only approximates when the job is submitted to the chain, not when it is included in a block and assigned to a transcoder.
```

Comment:
Uncles need to be handled by the client - transactions included in uncle blocks need to be included again in blocks on the main chain in order for them to be a part of the chain history. So it would be possible for an orchestrator to see a job transaction in a block, only to realize that the block is an uncle such that the job transaction does not yet exist on the main chain (but it could be included again in a subsequent block). We don't handle these types of cases right now, but we could implement a required number of confirmations to wait so clients can be more certain that a transaction was not included in an uncle (until we have finality with PoS one day so we don't need to wait for confirmations one day...)